src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -5265,4 +5265,514 @@ RppStatus color_cast_i8_i8_host_tensor(Rpp8s *srcPtr,
+                srcPtrTempR3 = srcPtrRowR3;
+                srcPtrTempG3 = srcPtrRowG3;
+                srcPtrTempB3 = srcPtrRowB3;
+
+                srcPtrTempR4 = srcPtrRowR4;
+                srcPtrTempG4 = srcPtrRowG4;
+                srcPtrTempB4 = srcPtrRowB4;
+
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount3 = 0;
+                for (; vectorLoopCount3 < bufferLength3; vectorLoopCount3++)
+                {
+                    dstPtrTemp[0] = *srcPtrTempR3;
+                    dstPtrTemp[1] = *srcPtrTempG3;
+                    dstPtrTemp[2] = *srcPtrTempB3;
+
+                    srcPtrTempR3++;
+                    srcPtrTempG3++;
+                    srcPtrTempB3++;
+                    dstPtrTemp += 3;
+                }
+
+                int vectorLoopCount4 = 0;
+                for (; vectorLoopCount4 < bufferLength4; vectorLoopCount4++)
+                {
+                    dstPtrTemp[0] = *srcPtrTempR4;
+                    dstPtrTemp[1] = *srcPtrTempG4;
+                    dstPtrTemp[2] = *srcPtrTempB4;
+
+                    srcPtrTempR4++;
+                    srcPtrTempG4++;
+                    srcPtrTempB4++;
+                    dstPtrTemp += 3;
+                }
+
+                srcPtrRowR3 += srcDescPtr->strides.hStride;
+                srcPtrRowG3 += srcDescPtr->strides.hStride;
+                srcPtrRowB3 += srcDescPtr->strides.hStride;
+                srcPtrRowR4 += srcDescPtr->strides.hStride;
+                srcPtrRowG4 += srcDescPtr->strides.hStride;
+                srcPtrRowB4 += srcDescPtr->strides.hStride;
+                dstPtrRow += dstDescPtr->strides.hStride;
+            }
+        }
+        // ricap without fused output-layout toggle (NHWC -> NHWC or NCHW -> NCHW)
+        else
+        {
+            Rpp32u alignedLength1 = bufferLength1 & ~15;
+            Rpp32u alignedLength2 = bufferLength2 & ~15;
+            Rpp32u alignedLength3 = bufferLength3 & ~15;
+            Rpp32u alignedLength4 = bufferLength4 & ~15;
+
+            for (int c = 0; c < layoutParams.channelParam; c++)
+            {
+                Rpp8u *srcPtrRow1, *srcPtrRow2, *srcPtrRow3, *srcPtrRow4, *dstPtrRow;
+                srcPtrRow1 = srcPtrChannel1;
+                srcPtrRow2 = srcPtrChannel2;
+                srcPtrRow3 = srcPtrChannel3;
+                srcPtrRow4 = srcPtrChannel4;
+                dstPtrRow = dstPtrChannel;
+
+                for (int i = 0; i < roiPtr1->xywhROI.roiHeight; i++)
+                {
+                    Rpp8u *srcPtrTemp1, *srcPtrTemp2, *srcPtrTemp3, *srcPtrTemp4, *dstPtrTemp; // put it outside the loop
+                    srcPtrTemp1 = srcPtrRow1;
+                    srcPtrTemp2 = srcPtrRow2;
+                    srcPtrTemp3 = srcPtrRow3;
+                    srcPtrTemp4 = srcPtrRow4;
+                    dstPtrTemp = dstPtrRow;
+
+                    //|---img-1---|----------img-2----------|
```

Comment:
Please move this comment to possibly the beginning of the function since its common to all sub-variants.
Remove any other instances of the same comment.
Keep the same standard like elsewhere while commenting. "//" followed by single space.
And add a single line header to the comment saying something like "// RICAP output image profile" so its clear what you are conveying.